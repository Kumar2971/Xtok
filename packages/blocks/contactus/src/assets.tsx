export const backArrow = require("../assets/backArrow.png");
export const imgClose= require('../assets/close.png');
export const imgLeftArrow = require("../../global_assets/leftArrow.png");
export const imgRightArrow = require("../../global_assets/rightArrow.png");