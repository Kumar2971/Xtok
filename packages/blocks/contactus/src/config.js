Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.httpGetMethod = "GET";
exports.httpPostMethod = "POST";
exports.httpDeleteMethod = "DELETE";
exports.formContentType = "multipart/form-data";
exports.postMethod = "POST";

exports.contactUsApiContentType = "application/json";
exports.getContactUsAPiEndPoint = "contact_us/contacts";
exports.getContactUsAPiEndPoint  = `bx_block_contact_us/report_problems`;
exports.errorTitle = "Error";
exports.errorAllFieldsAreMandatory = "All fields are mandatory.";
exports.errorEmailNotValid = "Email not valid.";
exports.phoneNumberError = "Phone number not valid"

exports.nameTitle = "Name";
exports.emailTitle = "Email";
exports.numberTitle = "Phone number";
exports.commentsTitle = "Comments";

exports.namePlaceHolder = "Enter Your Name";
exports.emailPlaceHolder = "Enter Your Email";
exports.numberPlaceHolder = "Enter Phone number";
exports.commentsPlaceHolder = "Enter Comments";

exports.labelTitleText = "contactus";
exports.labelBodyText = "contactus Body";

exports.btnExampleTitle = "CLICK ME";
exports.headerText = "Report a Problem";
exports.descriptionTitle = "Description";
exports.descriptionText = "Write a description and attach picture of the screens you faced an issue on. Try to be as descriptive as possible";
exports.addBtnTitle = "Add Photos";
exports.sendReportTitle = "Send Report";
exports.errorMSg = "Please add your description";
exports.imgerrorMSg = "You can't select more then 10 images";
exports.okText = "Ok"
// Customizable Area End