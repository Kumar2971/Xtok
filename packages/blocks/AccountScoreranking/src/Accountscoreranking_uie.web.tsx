//This an auto generated file for Artboard Name = Accountscoreranking And Supports ReactNative Ver. = 0.62
import React from 'react';
import { Dimensions, Image, SafeAreaView, StyleSheet, View } from 'react-native';

import MergeEngineUtilities from "../../utilities/src/MergeEngineUtilities";
import global_styles from "../../utilities/src/global_styles";
import { img23f5b10d71b5bccba285d46ac9825341cc080829, img5496f6a8bbff86b828368c46c94b63657895c52c } from './assets';
let screenWidth = Dimensions.get('window').width; 
let screenHeight = Dimensions.get('window').height; 
//Artboard Dimension 
let artBoardHeightOrg = 812; 
let artBoardWidthOrg = 375; 
//Re calculated Artboard Dimension 
let artBoardWidth =  isSameRatio() ? artBoardWidthOrg : screenWidth; 
let artBoardHeight =  isSameRatio() ? artBoardHeightOrg : screenHeight; 
// To check if Artboard and Device screen has same ratio 
function isSameRatio(): boolean {
    return artBoardWidthOrg / artBoardHeightOrg < 1 && screenWidth / screenHeight < 1 
}
 
//Top or Bottom nav spaces or any extra space occupied by os e.g Status bar, Notch 
let extraSpace = 0; 

 
 
export default class Accountscoreranking extends React.Component
{
	constructor(props:any)
	{ 
		super(props);
	} 

	state = {
	}

	componentDidMount() 
	{ 
		Dimensions.addEventListener('change', (e) => { 
			
			screenWidth = Dimensions.get('window').width; 
			screenHeight = Dimensions.get('window').height; 
			this.forceUpdate();
		});
	}

	render()
	{
		const local_styles = StyleSheet.create({
    AccountscorerankingView: {
        backgroundColor: "rgba(255, 249, 234, 1)",
        flex: 1,
        alignItems: "flex-start",
    },
    image_BitmapImage: {
        position: "absolute",
        width: MergeEngineUtilities.deviceBasedDynamicDimension(375, true, 1),
        height: MergeEngineUtilities.deviceBasedDynamicDimension(812, false, 1),
        marginLeft: MergeEngineUtilities.deviceBasedDynamicDimension(0, true, 1),
        marginTop: MergeEngineUtilities.deviceBasedDynamicDimension(0, false, 1),
        opacity: 1,
        resizeMode: "contain",
    },
})

const styles = {...global_styles, ...local_styles}
 
 
 	return (

		<SafeAreaView style={{"flex":1}}> 
		<View 
				style={styles.AccountscorerankingView}>
		<Image data-elementId='image_Bitmap' 
		   style={styles.image_BitmapImage}
		   source={img5496f6a8bbff86b828368c46c94b63657895c52c}
		/>
		<Image data-elementId='imagenav_dummy' 
		   style={styles.imagenav_dummyImageNav}
		   source={img23f5b10d71b5bccba285d46ac9825341cc080829}
		/>
			</View>
			</SafeAreaView>
	
		)
	}

}
