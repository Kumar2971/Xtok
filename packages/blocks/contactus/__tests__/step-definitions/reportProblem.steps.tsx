import React from "react";
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import ReportProblem from "../../src/ReportProblem";
// const navigation = require("react-navigation");

import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";
// export const configJSON = require("../../config.json");
import MessageEnum, {
  getName,
} from "../../../../framework/src/Messages/MessageEnum";
import * as utils from "../../../../framework/src/Utilities";
// import ImagePicker from "react-native-image-crop-picker";
import { View } from "react-native";

// contactUsApiCallId
const sendReportApiResponse: any = {"data":{"id":"62","type":"report_problem","attributes":{"description":"text added...","user":{"id":716,"first_name":null,"last_name":null,"full_phone_number":"919687806011","country_code":91,"phone_number":9687806011,"email":"test315@yopmail.com","activated":true,"device_id":null,"unique_auth_id":"tg343OBimg6lr58Qg6fC4Att","password_digest":"$2a$12$edfnVbafkYXPcrO2dDvlX.BFTS9suko6roPLqpr7.BS7.QxlGlG5i","created_at":"2023-04-05T04:18:01.470Z","updated_at":"2023-04-12T04:27:25.737Z","user_name":"Test315","platform":null,"user_type":null,"app_language_id":null,"last_visit_at":null,"is_blacklisted":false,"suspend_until":null,"status":"regular","stripe_id":null,"stripe_subscription_id":null,"stripe_subscription_date":null,"role_id":3,"full_name":"Test user","gender":null,"date_of_birth":"1992-04-04","age":null,"is_paid":false,"verified":true,"is_subscribed":false,"group_subscribed":false,"user_profile_data":null,"bio":null,"youtube":null,"instagram":null,"is_online":true,"nickname":null,"last_seen_at":"2023-04-12T04:27:25.688Z"},"images":[{"id":605,"url":"https://liketiktokapp-255799-ruby.b255799.dev.eastus.az.svc.builder.cafe/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbDBDIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--3f3700866bca9e777674532e8650bfbc642074d3/Screenshot%202023-04-04%20at%2012.31.29%20PM.png"},{"id":606,"url":"https://liketiktokapp-255799-ruby.b255799.dev.eastus.az.svc.builder.cafe/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbDRDIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--8f76446c10f01250637f85dc662eb7e41b427ef1/Screenshot%202023-03-29%20at%2011.16.09%20AM.png"}],"message":"Thanks for submit your feedback"}}};
const errorsResponse: any = {"errors":{"token": ""}};

const screenProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn()
  },
  id: "ReportProblem",
};

const feature = loadFeature("./__tests__/features/reportProblem-scenario.feature");

const obj = [
  {
    id:1,
    filename: "123.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    type: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:2,
    filename: "abc.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    type: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  }
]
const objwithImageType = [
  {
    id:1,
    filename: "123.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:2,
    filename: "abc.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  }
]

const imageArray = [
  {
    id:1,
    filename: "123.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    type: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:2,
    filename: "abc.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    type: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:3,
    filename: "123.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    type: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:4,
    filename: "abc.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:5,
    filename: "123.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:6,
    filename: "abc.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:7,
    filename: "123.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:8,
    filename: "abc.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:9,
    filename: "123.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  {
    id:10,
    filename: "abc.jpg",
    uri: '/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg',
    mime: "image/jpeg",
    name: "14E30C3D4-8621-4498-86E6-69BC4993E02E.jpg",
    path:"/Users/macm22/Library/Developer/CoreSimulator/Devices/93458AD8-AF00-4BED-9CEB-0D5F617D450D/data/Containers/Data/Application/2CD63ED3-7E14-42DA-A054-5326B36DC733/tmp/react-native-image-crop-picker/4E30C3D4-8621-4498-86E6-69BC4993E02E.jpg"
  },
  
]

const mockgetStorageData = jest.spyOn(utils, "getStorageData");
mockgetStorageData.mockResolvedValue("ar");

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
    jest.useFakeTimers();
  });

  test("User navigates to reportproblem", ({ given, when, then }) => {
    let ReportProblemWrapper: ShallowWrapper;
    let instance: ReportProblem;
    let flatListRenderWraper: ShallowWrapper;
    let itemObj: any;

    given("I am a User loading reportproblem", () => {
      //@ts-ignore
      ReportProblemWrapper = shallow(<ReportProblem {...screenProps} />);
    });

    when("I navigate to the reportproblem", () => {
      instance = ReportProblemWrapper.instance() as ReportProblem;
    });

    then("reportproblem will load with out errors", () => {
      expect(ReportProblemWrapper).toBeTruthy();
    });

    then("I can enter comment with out error", () => {
      let txtInputComponent = ReportProblemWrapper.findWhere(
        node => node.prop("testID") === "txtInputDescription"
      );
      txtInputComponent.simulate("changeText", "Your description here.");
      txtInputComponent.simulate("submitEditing");
    });
   
    then("I will upload Images without errors", () => {
      const { ImageSourceviewarray } = instance.state;
      itemObj = {
        item: ImageSourceviewarray[0] || [],
        index: 0
      }

      let buttonComponent = ReportProblemWrapper.findWhere(
        node => node.prop("testID") === "addphphotosBtn"
      );
      buttonComponent.simulate("press");

      let imageFlatlist = ReportProblemWrapper.findWhere(
        node => node.prop("testID") === "imageFlatlist"
      );
      imageFlatlist.props().renderItem(itemObj);
      imageFlatlist.props().keyExtractor("1");

      flatListRenderWraper = shallow(
        //@ts-ignore
        <View>
          {instance.renderImage(itemObj)};
        </View>
      );
    });

    then("I will delete Images without errors", () => {
      const {ImageSourceviewarray} = instance.state;
      itemObj = {
        item: ImageSourceviewarray[0] || [],
        index: 0
      }

      let buttonComponent = flatListRenderWraper.findWhere(
        node => node.prop("testID") === "deleteBtn"
      );
      buttonComponent.simulate("press");
    });

    then("I can select send report button with out error", () => {
      let buttonComponent = ReportProblemWrapper.findWhere(
        node => node.prop("testID") === "sendReportBtn"
      );
      buttonComponent.simulate("press");

      //send report apicall
      const sendReportProblemApiCallId = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      )
      sendReportProblemApiCallId.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        sendReportProblemApiCallId.messageId
      )
      sendReportProblemApiCallId.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        JSON.parse(JSON.stringify(sendReportApiResponse))
      )
      instance.sendReportProblemApiCallId = sendReportProblemApiCallId.messageId;
      runEngine.sendMessage('Unit Test', sendReportProblemApiCallId)

      instance.setReportModal(true);
      instance.goToNext();
    });

    then("I can select send report with error", () => { 
      let txtInputComponent = ReportProblemWrapper.findWhere(
        node => node.prop("testID") === "txtInputDescription"
      );
      txtInputComponent.simulate("changeText", "");

      let buttonComponent = ReportProblemWrapper.findWhere(
        node => node.prop("testID") === "sendReportBtn"
      );
      buttonComponent.simulate("press");
      //send report apicall
      const sendReportProblemApiCallId = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      )
      sendReportProblemApiCallId.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        sendReportProblemApiCallId.messageId
      )
      sendReportProblemApiCallId.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        {"errors":{"token": ""}}
      )
      instance.sendReportProblemApiCallId = sendReportProblemApiCallId.messageId;
      runEngine.sendMessage('Unit Test', sendReportProblemApiCallId)


       buttonComponent = ReportProblemWrapper.findWhere(
          (node) => node.prop("testID") === "submitModal"
        );
      buttonComponent.simulate("popup");
    });

    then('I can close popup with ok button', () => {
      const goToNextMock =jest.spyOn(instance,"goToNext")
      let buttonComponent = ReportProblemWrapper.findWhere(
          (node) => node.prop("testID") === "okBtn"
        );
      buttonComponent.simulate("press");
      let submitModal = ReportProblemWrapper.findWhere(
        (node) => node.prop("testID") === "submitModal"
      );
      submitModal.simulate("requestClose")  
      expect(goToNextMock).toHaveBeenCalledTimes(1);
  });

    then("I can select send report button with image type", () => { 
      instance = ReportProblemWrapper.instance() as ReportProblem;
    });

    then("I can leave the screen with out errors", () => {
      let btnGoBack = ReportProblemWrapper.findWhere(
        (node:any) => node.prop("testID") === "btnGoBack"
      );
      btnGoBack.simulate("press");
      instance.handleErrorResponse(errorsResponse);
      instance.componentWillUnmount();
      expect(ReportProblemWrapper).toBeTruthy();
    });
  });

  test("User navigates to reportproblem with error", ({ given, when, then }) => {
    let ReportProblemWrapper: ShallowWrapper;
    let instance: ReportProblem;

    given("I am a User loading reportproblem with error", () => {
      //@ts-ignore
      ReportProblemWrapper = shallow(<ReportProblem {...screenProps} />);
    });

    when("I navigate to the reportproblem with error", () => {
      instance = ReportProblemWrapper.instance() as ReportProblem;
    });

    then("reportproblem will load with errors", () => {
      let txtInputComponent = ReportProblemWrapper.findWhere(
        node => node.prop("testID") === "txtInputDescription"     
      );
      txtInputComponent.simulate("changeText", "");
      expect(ReportProblemWrapper).toBeTruthy();
    });
  });

});
