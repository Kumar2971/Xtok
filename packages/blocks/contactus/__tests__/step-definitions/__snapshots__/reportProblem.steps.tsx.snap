// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reportproblem User navigates to reportproblem 1`] = `
<KeyboardAvoidingView
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "height": 72.46376811594203,
            "paddingHorizontal": 27.17391304347826,
            "width": "100%",
          },
          Object {
            "borderBottomColor": "#B7BBC0",
            "borderBottomWidth": 0.5,
          },
        ]
      }
    >
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "height": 38.5,
            "width": 38.5,
          }
        }
        testID="btnGoBack"
      >
        <Image
          source={
            Object {
              "testUri": "../../../packages/blocks/global_assets/leftArrow.png",
            }
          }
          style={
            Object {
              "height": 45.289855072463766,
              "resizeMode": "contain",
              "width": 45.289855072463766,
            }
          }
        />
      </ForwardRef>
      <Text
        style={
          Array [
            Object {
              "fontSize": 32.608695652173914,
              "fontWeight": "bold",
              "textAlign": "center",
              "width": 641.304347826087,
            },
          ]
        }
      >
        Report a problem
      </Text>
    </View>
    <TouchableWithoutFeedback
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffffff",
          "flex": 1,
          "marginLeft": "auto",
          "marginRight": "auto",
          "maxWidth": 650,
          "padding": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffffff",
            "flex": 1,
            "marginLeft": "auto",
            "marginRight": "auto",
            "maxWidth": 650,
            "padding": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#9E9E9E",
                "fontFamily": "Montserrat-Medium",
                "fontSize": 27,
                "textTransform": "uppercase",
              },
              false,
            ]
          }
        >
          Description
        </Text>
        <Text
          style={
            Object {
              "color": "#9E9E9E",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 23,
              "marginVertical": 16,
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Write a description and attach pictures of the screens you faced an issue on. Try to be as descriptive as possible
        </Text>
        <TextInput
          allowFontScaling={true}
          maxLength={200}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Write here..."
          placeholderTextColor="#B7BBC0"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#FAFAFA",
                "borderColor": "#FFFFFF",
                "borderRadius": 19,
                "borderWidth": 0.5,
                "fontFamily": "Montserrat-Regular",
                "height": "20%",
                "marginVertical": 16,
                "paddingLeft": 29,
                "paddingTop": 29,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {
                "textAlign": "left",
              },
            ]
          }
          testID="txtInputDescription"
          underlineColorAndroid="transparent"
          value="Your description here."
        />
        <View />
        <View>
          <FlatList
            bounces={false}
            data={Array []}
            decelerationRate="fast"
            disableVirtualization={false}
            horizontal={true}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
            testID="imageFlatlist"
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
          <View />
        </View>
        <View
          style={
            Object {
              "bottom": 31.5,
              "left": 16,
              "position": "absolute",
              "right": 0,
              "width": "100%",
            }
          }
        >
          <mockConstructor
            buttonStyle={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#FFC925",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "height": 96,
                },
              ]
            }
            containerStyle={
              Object {
                "marginTop": 16,
              }
            }
            onPress={[Function]}
            testID="addphphotosBtn"
            title="Add Photos"
            titleStyle={
              Object {
                "color": "#000",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 32.608695652173914,
                "fontWeight": "bold",
              }
            }
          />
          <mockConstructor
            buttonStyle={
              Array [
                Object {
                  "backgroundColor": "#FFC925",
                  "borderRadius": 50,
                  "height": 96,
                },
              ]
            }
            containerStyle={
              Object {
                "marginTop": 16,
              }
            }
            disabled={true}
            loading={false}
            onPress={[Function]}
            testID="sendReportBtn"
            title="Send Report"
            titleStyle={
              Object {
                "color": "#000",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 32.608695652173914,
                "fontWeight": "bold",
              }
            }
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
    <View
      style={Object {}}
    >
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        testID="submitModal"
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.2)",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#FFC925",
                "borderRadius": 20,
                "borderWidth": 1,
                "marginHorizontal": "5%",
                "padding": 30,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 27,
                  }
                }
              >
                Thanks for submitting your feedback
              </Text>
            </View>
            <View
              style={Object {}}
            >
              <ForwardRef
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFC925",
                    "borderRadius": 96,
                    "height": 96,
                    "justifyContent": "center",
                    "marginTop": 38.5,
                  }
                }
                testID="okBtn"
              >
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 32.608695652173914,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Ok
                </Text>
              </ForwardRef>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </ForwardRef(SafeAreaView)>
</KeyboardAvoidingView>
`;
